elide-core/src/main/java/com/yahoo/elide/parsers/expression/PermissionToFilterExpressionVisitor.java

```diff
@@ -157,4 +163,42 @@ private Operator operator(FilterExpression expression) {
     public FilterExpression visitPAREN(ExpressionParser.PARENContext ctx) {
         return visit(ctx.expression());
     }
+
+    /**
+     * normalizeNegation function serves to push down NotFilterexpression to the predicate level.
+     * @param fe FilterExpression that need to normalize
+     * @return A normalized version of input
+     */
+    public static FilterExpression normalizeNegation(FilterExpression fe) {
+        if (fe instanceof NotFilterExpression) {
+            FilterExpression nfe = ((NotFilterExpression) fe).getNegated();
+            if (nfe instanceof AndFilterExpression) {
+                return new OrFilterExpression(
+                        normalizeNegation(new NotFilterExpression(((AndFilterExpression) nfe).getLeft())),
+                        normalizeNegation(new NotFilterExpression(((AndFilterExpression) nfe).getRight())));
+            } else if (nfe instanceof OrFilterExpression) {
+                return new AndFilterExpression(
+                        normalizeNegation(new NotFilterExpression(((OrFilterExpression) nfe).getLeft())),
+                        normalizeNegation(new NotFilterExpression(((OrFilterExpression) nfe).getRight())));
+            } else if (nfe instanceof FilterPredicate) {
+                ((FilterPredicate) nfe).negate();
+                return nfe;
+            } else if (nfe instanceof NotFilterExpression) {
+                return normalizeNegation(((NotFilterExpression) nfe).getNegated());
+            } else {
+                return nfe;
+            }
+        } else if (fe instanceof AndFilterExpression) {
+            return new AndFilterExpression(
+                    normalizeNegation(((AndFilterExpression) fe).getLeft()),
+                    normalizeNegation(((AndFilterExpression) fe).getRight()));
+        } else if (fe instanceof OrFilterExpression) {
+            return new OrFilterExpression(
+                    normalizeNegation(((OrFilterExpression) fe).getLeft()),
+                    normalizeNegation(((OrFilterExpression) fe).getRight()));
+        } else if (fe instanceof FilterPredicate) {
```

Comment:
This conditional is unnecessary. Just let it fall through to the final return statement since they're the same.